subsys/net/openthread/rpc/common/ot_rpc_common.c

```diff
@@ -269,3 +272,256 @@ bool ot_rpc_decode_message_info(struct nrf_rpc_cbor_ctx *ctx, otMessageInfo *aMe
+		return false;
+	}
+
+	if (!zcbor_uint_decode(ctx->zs, &config->mEnterpriseNumber,
+			       sizeof(config->mEnterpriseNumber))) {
+		return false;
+	}
+
+	if (!zcbor_bstr_decode(ctx->zs, &zst)) {
+		return false;
+	}
+
+	config->mServiceDataLength = zst.len;
+	memcpy(config->mServiceData, zst.value, zst.len);
+
+	if (!zcbor_bool_decode(ctx->zs, (bool *)&tmp)) {
+		return false;
+	}
+
+	config->mServerConfig.mStable = tmp;
+
+	if (!zcbor_bstr_decode(ctx->zs, &zst)) {
+		return false;
+	}
+
+	config->mServerConfig.mServerDataLength = zst.len;
+	memcpy(config->mServerConfig.mServerData, zst.value, zst.len);
+
+	if (!zcbor_uint_decode(ctx->zs, &config->mServerConfig.mRloc16,
+			       sizeof(config->mServerConfig.mRloc16))) {
+		return false;
+	}
+
+	return true;
+}
+
+bool ot_rpc_encode_border_router_config(struct nrf_rpc_cbor_ctx *ctx,
+					const otBorderRouterConfig *config)
+{
+	uint8_t tmp;
+
+	if (config == NULL) {
+		return zcbor_nil_put(ctx->zs, NULL);
+	}
+
+	if (!zcbor_bstr_encode_ptr(ctx->zs, (const char *)config->mPrefix.mPrefix.mFields.m8,
+				   OT_IP6_ADDRESS_SIZE)) {
+		return false;
+	}
+
+	if (!zcbor_uint_encode(ctx->zs, &config->mPrefix.mLength,
+			       sizeof(config->mPrefix.mLength))) {
+		return false;
+	}
+
+	tmp = config->mPreference;
+	if (!zcbor_uint_encode(ctx->zs, &tmp, sizeof(tmp))) {
+		return false;
+	}
+
+	if (!zcbor_bool_put(ctx->zs, config->mPreferred)) {
+		return false;
+	}
+
+	if (!zcbor_bool_put(ctx->zs, config->mSlaac)) {
+		return false;
+	}
+
+	if (!zcbor_bool_put(ctx->zs, config->mDhcp)) {
+		return false;
+	}
+
+	if (!zcbor_bool_put(ctx->zs, config->mConfigure)) {
+		return false;
+	}
+
+	if (!zcbor_bool_put(ctx->zs, config->mDefaultRoute)) {
+		return false;
+	}
+
+	if (!zcbor_bool_put(ctx->zs, config->mOnMesh)) {
+		return false;
+	}
+
+	if (!zcbor_bool_put(ctx->zs, config->mStable)) {
+		return false;
+	}
+
+	if (!zcbor_bool_put(ctx->zs, config->mNdDns)) {
+		return false;
+	}
+
+	if (!zcbor_bool_put(ctx->zs, config->mDp)) {
+		return false;
+	}
+
+	if (!zcbor_uint_encode(ctx->zs, &config->mRloc16, sizeof(config->mRloc16))) {
+		return false;
+	}
+
+	return true;
+}
+
+bool ot_rpc_decode_border_router_config(struct nrf_rpc_cbor_ctx *ctx, otBorderRouterConfig *config)
+{
+	struct zcbor_string zst;
+	uint8_t tmp;
+
+	if (zcbor_nil_expect(ctx->zs, NULL)) {
+		memset(config, 0, sizeof(otBorderRouterConfig));
+		return true;
+	}
+
+	if (ctx->zs->constant_state->error != ZCBOR_ERR_WRONG_TYPE) {
+		return false;
+	}
+
+	zcbor_pop_error(ctx->zs);
+
+	if (!zcbor_bstr_decode(ctx->zs, &zst)) {
+		return false;
+	}
+
+	memcpy(config->mPrefix.mPrefix.mFields.m8, zst.value, zst.len);
+
+	if (!zcbor_uint_decode(ctx->zs, &config->mPrefix.mLength,
+			       sizeof(config->mPrefix.mLength))) {
+		return false;
+	}
+
+	if (!zcbor_uint_decode(ctx->zs, &tmp, sizeof(tmp))) {
+		return false;
+	}
+
+	config->mPreference = tmp;
+
+	if (!zcbor_bool_decode(ctx->zs, (bool *)&tmp)) {
```

Comment:
Can we avoid casting `uint8_t*` to `bool*` (which assumes a given bool representation)?